pkg/operator/controller/garden/reference/add.go

```diff
@@ -40,9 +42,24 @@ func Predicate(oldObj, newObj client.Object) bool {
 		return false
 	}
 
+	oldProviderSecretNames := map[string]string{}
+	newProvidersSecretNames := map[string]string{}
+	if oldGarden.Spec.DNS != nil {
+		for _, provider := range oldGarden.Spec.DNS.Providers {
+			oldProviderSecretNames[provider.Name] = provider.SecretRef.Name
+		}
+	}
+	if newGarden.Spec.DNS != nil {
+		for _, provider := range newGarden.Spec.DNS.Providers {
+			newProvidersSecretNames[provider.Name] = provider.SecretRef.Name
+		}
+	}
+	dnsSecretsChanged := !reflect.DeepEqual(oldProviderSecretNames, newProvidersSecretNames)
+
 	return kubeAPIServerAuditPolicyConfigMapChanged(oldGarden.Spec.VirtualCluster.Kubernetes.KubeAPIServer, newGarden.Spec.VirtualCluster.Kubernetes.KubeAPIServer) ||
 		gardenerAPIServerAuditPolicyConfigMapChanged(oldGarden.Spec.VirtualCluster.Gardener.APIServer, newGarden.Spec.VirtualCluster.Gardener.APIServer) ||
 		etcdBackupSecretChanged(oldGarden.Spec.VirtualCluster.ETCD, newGarden.Spec.VirtualCluster.ETCD) ||
+		dnsSecretsChanged ||
```

Comment:
Maybe just cosmetic but could you make it consistent to other lines and use a function instead?